pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/session/PillarboxMediaController.kt

```diff
@@ -233,6 +235,9 @@ open class PillarboxMediaController internal constructor() : PillarboxPlayer {
 
     internal fun setMediaController(mediaController: MediaController) {
         this.mediaController = mediaController
+        listeners = ListenerSet(mediaController.applicationLooper, Clock.DEFAULT) { listener, flags ->
```

Comment:
To avoid potential outdated events to be send, I suggest to first call `listeners?.release()`, before recreating a `ListenerSet`.